samcli/commands/_utils/options.py

```diff
@@ -680,6 +682,25 @@ def resolve_s3_click_option(guided):
     )
 
 
+def hook_package_id_click_option(force_prepare=True, invalid_coexist_options=None):
```

Comment:
for force_prepare, I chose to set the default to `True` as it is only `sam local` commands that does not require running  the prepare hook, but `sam sync` and `sam build` commands require running the prepare hook.

Reply:
I decided to make `invalid_coexists_options` to be customizable, to be felixaeble in case if we decided  to change one command later and to add some new option for one command that should not be used in case of other IaCs support.